scanpipe/pipes/d2d.py

```diff
@@ -614,12 +616,23 @@ def _map_about_file_resource(project, about_file_resource, to_resources):
         # Cannot map anything without the about_resource value.
         return
 
+    ignored_resources = []
+    if extra_data := package_data.get("extra_data"):
+        ignored_resources = extra_data.get("ignored_resources")
+
     # Fetch all resources that are covered by the .ABOUT file.
     codebase_resources = to_resources.filter(path__contains=f"/{filename.lstrip('/')}")
     if not codebase_resources:
         # If there's nothing to map on the ``to/`` do not create the package.
         return
 
+    # Ignore resources for paths in `ignored_resources` attribute
+    if ignored_resources:
+        lookups = Q()
+        for resource_path in ignored_resources:
+            lookups |= Q(**{"path__contains": resource_path})
+        codebase_resources = codebase_resources.filter(~lookups)
```

Comment:
@AyanSinhaMahapatra one clarification: `ignored_resources` is a list of exact paths right?
This code looks like it deals with patterns (ie: `contains`) more than exact resource paths.
I want to make sure this is the intended behavior.
Could you point me to the spec of the AboutCode `ignored_resources` field?

Reply:
@tdruez it is a list of paths which should be ignored, but this also means ignoring anything under that path, i.e. if the path is a directory, then all the files/sub-folders are to be ignored too. So it's not just list of exact paths which should be ignored.
See the aboutcode spec at:
https://aboutcode-toolkit.readthedocs.io/en/latest/specification.html#referencing-the-file-or-directory-documented-by-an-about-file say:
```
In this example, the ABOUT file documents a whole sub-directory, with some sub-paths under the directory ignored:

    about_resource: linux-kernel-2.6.23
    ignored_resources: linux-kernel-2.6.23/Documentation

```
Also at https://aboutcode-toolkit.readthedocs.io/en/latest/specification.html#optional-information-fields:
```
ignored_resources: A list of paths under the about_resource path, which are not documented in the ABOUT file, and the information in the ABOUT file does not apply to these subpaths.
```

Reply:
@AyanSinhaMahapatra thanks for the precisions.
Is that behavior "files/sub-folders" covered by the unit tests?

Reply:
> Is that behavior "files/sub-folders" covered by the unit tests?

Not entirely, as the package used for the tests did not have that much sub-folders to be able to do this properly.
But since `contains` is used I didn't think it would require to be added as a test. I have tested with other bigger packages locally and this works well with a list of paths which are directories, and everything below them are ignored from being assigned to the package.

Neither the function `map_about_files` nor `_map_about_file_resource` has unit tests before, so I didn't add more unit tests there too. So we just have a full pipeline run test instead, and I modified the same.

I've added this to the about file:
```
about_resource: flume-ng-node-1.9.0.jar
ignored_resources:
  - flume-ng-node-1.9.0.jar-extract/org/apache/flume/node/ConfigurationProvider.class
```

And this results in excluding the file from being assigned to the package: https://github.com/nexB/scancode.io/blob/f6981cf029de6276e8ec2cc5240ab3642785c2c8/scanpipe/tests/data/d2d/about_files/expected.json#L1258

If you think that's better, I can revert the existing test case to whatever it was before, and add a new test case with more files/sub-folders too. Please advise on the same.